### PR TITLE
Playwright initial setup

### LINKED ISSUE #872: FEAT: Add capability for e2e tests
**Please Describe The Problem To Be Solved**
Need to add capability for executing automated e2e tests to save on manual testing efforts

**(Optional): Suggest A Solution**
Add playwright framework using Typescript

### PR DESCRIPTION
**Pull request by anupdhiran**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #872 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Ran log in test using different combinations and verified that the admin.json and user.json files were getting modified with respective values and the test was able to launch without requiring to log in again.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,4 +29,8 @@ yarn-error.log
 /public/env-config.js
 env-config.js
 .idea/
+/test-results/
+/playwright-report/
+/playwright/.cache/
+playwright/.auth
```
Comment (sohel-rp) [inline .gitignore:35]: missing /

File: e2e-tests/auth.setup.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { test as setup } from '@playwright/test';
```
Comment (sohel-rp) [inline e2e-tests/auth.setup.ts:1]: why are we doing this?

Comment (sohel-rp) [reply e2e-tests/auth.setup.ts:1]: Got it.

File: playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import { defineConfig, devices } from '@playwright/test';
+
+/**
+ * Read environment variables from file.
+ * https://github.com/motdotla/dotenv
+ */
+// require('dotenv').config();
```
Comment (sohel-rp) [inline playwright.config.ts:7]: please remove commented if not required.

File: playwright.config.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    // {
+    //   name: 'firefox',
+    //   use: { ...devices['Desktop Firefox'] },
+    // },
+
+    // {
+    //   name: 'webkit',
+    //   use: { ...devices['Desktop Safari'] },
+    // },
```
Comment (sohel-rp) [inline playwright.config.ts:66]: same here

File: playwright.config.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  ],
+
+  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
+  // outputDir: 'test-results/',
+
+  /* Run your local dev server before starting the tests */
+  // webServer: {
+  //   command: 'npm run start',
```
Comment (sohel-rp) [inline playwright.config.ts:75]: same here

File: tests-examples/demo-todo-app.spec.ts [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+  await page.goto('https://demo.playwright.dev/todomvc');
+});
+
+const TODO_ITEMS = [
+  'buy some cheese',
+  'feed the cat',
+  'book a doctors appointment'
+];
+
+test.describe('New Todo', () => {
```
Comment (sohel-rp) [inline tests-examples/demo-todo-app.spec.ts:13]: same here

Comment (sohel-rp) [reply tests-examples/demo-todo-app.spec.ts:13]: import describe directly

File: playwright.config.ts (hunk 5 above)
Comment (anupdhiran) [reply playwright.config.ts:75]: this also will be useful whenever we want to record

File: tests-examples/demo-todo-app.spec.ts (hunk 6 above)
Comment (anupdhiran) [reply tests-examples/demo-todo-app.spec.ts:13]: These are demo files which are auto generated

File: playwright.config.ts (hunk 5 above)
Comment (sohel-rp) [reply playwright.config.ts:69]: ok

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**